code_review/16-09-20_s2s_util.py

```diff
@@ -0,0 +1,129 @@
+    """
+    if sample.startswith(("SRR", "DRR", "ERR")):
+        idxs = df.index[df.run_accession == sample].tolist()
+        assert len(idxs) == 1, f"sample {sample} with idxs: {idxs}"
+    elif sample.startswith("SRX"):
+        idxs = df.index[df.experiment_accession == sample].tolist()
+        assert len(idxs) >= 1, len(idxs)
+    else:
+        assert False
+    return idxs
+
+
+def samples2metadata(samples: List[str], config: dict) -> dict:
+    """
+    Get the required info to continue a seq2science run from a list of samples.
+
+    - If a sample already exists locally, we only want to know if it is paired-end or single-end.
+    - If a sample does not exist locally
+      - find its corresponding SRX number and all runs that belong to it,
+      - check if they all have the same layout, if not, crash
+      - see if we can download the runs from ena
+
+    output:
+        dict(
+            "GSM1234": {"layout": "PAIRED",
+                         "runs": ["SRR1234", "SRR4321"],
+                         "ena_fastq_http": {
+                            "SRR1234": [...],
+                            "SRR4321": None
+                            },
+                         "ena_fastq_ftp": ["..."],
+
+            "SRR5678": {"Layout": "SINGLE",
+                        "runs": ["SRR5678"],
+                        ena_fastq_http: None,
+                        ena_fastq_ftp: [...],
+            ...
+        )
+    """
+    # start with empty dictionary which we fill out later
+    sampledict = {sample: dict() for sample in samples}
+
+    # fill out the sampledict for the local samples, and store the public samples for later
+    public_samples = []
+    for sample in samples:
+        if os.path.exists(expand(f'{{fastq_dir}}/{sample}.{{fqsuffix}}.gz', **config)[0]):
+            sampledict[sample]["layout"] = "SINGLE"
+        elif all(os.path.exists(path) for path in expand(f'{{fastq_dir}}/{sample}_{{fqext}}.{{fqsuffix}}.gz', **config)):
+            sampledict[sample]["layout"] = "PAIRED"
+        elif sample.startswith(('GSM', 'SRX', 'SRR', 'ERR', 'DRR')):
+            public_samples.append(sample)
+        else:
+            raise ValueError(f"\nsample {sample} was not found..\n"
+                             f"We checked for SE file:\n"
+                             f"\t{config['fastq_dir']}/{sample}.{config['fqsuffix']}.gz \n"
+                             f"and for PE files:\n"
+                             f"\t{config['fastq_dir']}/{sample}_{config['fqext1']}.{config['fqsuffix']}.gz \n"
+                             f"\t{config['fastq_dir']}/{sample}_{config['fqext2']}.{config['fqsuffix']}.gz \n"
+                             f"and since the sample did not start with either GSM, SRX, SRR, ERR, and DRR we "
+                             f"couldn't find it online..\n")
+
+    if len(public_samples) == 0:
+        return sampledict
+
+    # only continue with public samples
+    db = pysradb.SRAweb()
```

Comment:
Are you loading a database from pysradb here? Because I think db is then not so clear. db_SRA maybe?